src/Symbiote/Components/ComponentService.php

```diff
@@ -121,75 +106,105 @@ public function generateTemplateCode(array $res, $parser)
+                        break;
+                    }
             }
-            $phpCode .= "\$_props['".$propertyName."'] = \SilverStripe\Core\Injector\Injector::inst()->get({$thisClassIdent})->createProperty('".$propertyName."', \$_props['".$propertyName."']);\n";
-            $arguments[$propertyName] = $phpCode;
+            throw new SSTemplateParseException('JSON ' . json_last_error_msg() . "\n" . $jsonString, $parser);
         }
 
-        // Output "children" php code
-        $value = isset($res['Children']['php']) ? $res['Children']['php'] : '';
-        if ($value) {
-            if (isset($arguments['children'])) {
-                throw new SSTemplateParseException('Cannot use "children" as a property name and have inner HTML.', $parser);
+        // process data into buffer
+        $buffer = '';
+        foreach ($jsonData as $name => $value) {
+            // handle nested data
+            if (is_array($value)) {
+                $value = self::exportNestedDataForTemplates($value);
+            }
+            // handle strings
+            else if (is_string($value)) {
+                $value = '"' . $value . '"';
             }
-            $value = "\$_props['children'] = '';\n".$value;
-            $value = str_replace("\$val .=", "\$_props['children'] .=", $value);
-            $value .= "\$_props['children'] = \SilverStripe\ORM\FieldType\DBField::create_field('HTMLFragment', \$_props['children']);\n";
-            $arguments['children'] = $value;
+            $buffer .= self::Prop2PHP($name, $value);
         }
 
-        // Output PHP code for setting properties
-        $result = "\$_props = array();\n";
-        foreach ($arguments as $propertyName => $phpCode) {
-            $result .= $phpCode;
+        return $buffer;
+    }
+
+    private static function handleChildHTML($data, $properties, $parser)
+    {
+        if (isset($data['Children']['php'])) {
+            // handle children property collision
+            if (isset($properties['children'])) {
+                throw new SSTemplateParseException('Cannot use "children" as a property name and have inner HTML.', $parser);
+            }
+
+            // construct php
+            $value = $data['Children']['php'];
+            $php = "\$_props['children'] = '';\n" . $value;
+            $php = str_replace("\$val .=", "\$_props['children'] .=", $php);
+            $php .= "\$_props['children'] = SilverStripe\ORM\FieldType\DBField::create_field('HTMLText', \$_props['children']);\n";
+
+            return $php;
         }
-        $result .= <<<PHP
-\$val .= \SilverStripe\Core\Injector\Injector::inst()->get({$thisClassIdent})->renderComponent('$componentName', \$_props, \$scope);
-unset(\$_props);
-PHP;
-        if (self::DEBUG_JSON_CODE_OUTPUT &&
-            $debugHasUsedJSON) {
-            var_dump($result);
-            throw new Exception('Debug stop.');
+        return '';
+    }
+
+    private static function Prop2PHP($name, $value)
```

Comment:
Can this start with a lowercase `p` to be consistent?

Reply:
Done.